modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -3934,23 +3970,33 @@ protected void encodeAttribute(AttributeDescriptor attribute, Query query, Strin
         }
     }
 
-    protected void encodeFunction( String function, AttributeDescriptor att, Query query, StringBuffer sql ) {
-        sql.append(function).append("(");
+    protected void encodeFunction( String function, Expression expression, Query query, StringBuffer sql , SimpleFeatureType featureType) {
+        sql.append(getFunctioName(function)).append("(");
+        AttributeDescriptor att = extractAggregateAttribute(expression, featureType);
         if ( att == null ) {
             sql.append( "*" );
         }
-        else {
+        else if (expression instanceof PropertyName){
             if ( att instanceof GeometryDescriptor ) {
                 encodeGeometryColumn((GeometryDescriptor)att, sql,query.getHints());
             }
             else {
                 dialect.encodeColumnName( att.getLocalName(), sql);
             }
+        } else if(expression instanceof FunctionExpression) {
+            dialect.encodeFunction(getFunctioName(((FunctionExpression)expression).getName()), att.getLocalName(), sql);
+        } else {
+            sql.append( "*" );
         }
         
         sql.append(")");
     }
     
+    private String getFunctioName(String function) {
```

Comment:
Confused, how do we reach here with a function name that is not supported? Wouldn't that be an error?

Reply:
You are right, it cannot happen, I will refactor the method